Threads/UARTCommsController.cpp

```diff
@@ -6,330 +6,337 @@
+  if (pHUART->Instance == USART7) {
+    s_pUARTController7 = this;
+  }
+#endif
+#ifdef USART8
+  if (pHUART->Instance == USART8) {
+    s_pUARTController8 = this;
+  }
+#endif
+  m_bTestVariable = true;
 }
 
-UARTCommsController::~UARTCommsController() {
-}
+UARTCommsController::~UARTCommsController() {}
 
-void UARTCommsController::Run(){
-	if(m_bDebugChannel){
-		if(__HAL_RCC_GET_FLAG(RCC_FLAG_IWDGRST) != RESET){
-			SendReply("Independent watchdog reset.");
-		}
-		if(__HAL_RCC_GET_FLAG(RCC_FLAG_PINRST)){
-			SendReply("Pin reset.");
-		}
-		if(__HAL_RCC_GET_FLAG(RCC_FLAG_BORRST) != RESET){
-			SendReply("Brown-out reset.");
-		}
-		if(__HAL_RCC_GET_FLAG(RCC_FLAG_SFTRST)){
-			SendReply("Software reset.");
-		}
-		if(__HAL_RCC_GET_FLAG(RCC_FLAG_WWDGRST) != RESET){
-			SendReply("Window watchdog reset.");
-		}
-		if(__HAL_RCC_GET_FLAG(RCC_FLAG_LPWRRST) != RESET){
-			SendReply("Low power reset.");
-		}
-		SendReply("Ready!");
-	}
-	/* reset flag */
-	__HAL_RCC_CLEAR_RESET_FLAGS();
+void UARTCommsController::Run() {
+  if (m_bDebugChannel) {
+    if (__HAL_RCC_GET_FLAG(RCC_FLAG_IWDGRST) != RESET) {
+      SendReply("Independent watchdog reset.");
+    }
+    if (__HAL_RCC_GET_FLAG(RCC_FLAG_PINRST)) {
+      SendReply("Pin reset.");
+    }
+    if (__HAL_RCC_GET_FLAG(RCC_FLAG_BORRST) != RESET) {
+      SendReply("Brown-out reset.");
+    }
+    if (__HAL_RCC_GET_FLAG(RCC_FLAG_SFTRST)) {
+      SendReply("Software reset.");
+    }
+    if (__HAL_RCC_GET_FLAG(RCC_FLAG_WWDGRST) != RESET) {
+      SendReply("Window watchdog reset.");
+    }
+    if (__HAL_RCC_GET_FLAG(RCC_FLAG_LPWRRST) != RESET) {
+      SendReply("Low power reset.");
+    }
+    SendReply("Ready!");
+  }
+  /* reset flag */
+  __HAL_RCC_CLEAR_RESET_FLAGS();
 
-	HAL_UART_Receive_IT(m_pHUART, &m_IncomingRxChar, 1);
-	for(;;){
-		/* Check for incoming data */
-		char incoming;
-		bool bMoreData = false;
-		bool bCommandReady = false;
-		if(!m_RxBuffer.IsEmpty()){
-			do{
-			    taskENTER_CRITICAL();
-				bMoreData = m_RxBuffer.Get(&incoming);
-				taskEXIT_CRITICAL();
-				bCommandReady = ParseCommand(&incoming);
-			}while(bMoreData & !bCommandReady);
-		}
-		/* If new command is ready, send to all controllers */
-		if(bCommandReady){
-			bool bCommandRecognised = false;
-			char reply[200];
-			unsigned int controller = 0;
-			do{
-				*reply = '\0';
-				bCommandRecognised = (m_pControllers[controller])->NewCommand((char*)m_NewCommand, reply);
-				if(bCommandRecognised){
-					SendReply(reply);
-				}
-			}while(m_pControllers[++controller] != NULL && !bCommandRecognised);
-			/* if command not recognised send generic reply. */
-			if(!bCommandRecognised){
-				SendReply("Command not recognised.");
-			}
-		}
+  HAL_UART_Receive_IT(m_pHUART, &m_IncomingRxChar, 1);
+  for (;;) {
+    /* Check for incoming data */
+    char incoming;
+    bool bMoreData = false;
+    bool bCommandReady = false;
+    if (!m_RxBuffer.IsEmpty()) {
+      do {
+        taskENTER_CRITICAL();
```

Comment:
this is dangerous. Do not use this type of call, because, it will cause memory leak.

Reply:
OK. It shall be fixed in next commit.